include/linux/uaccess.h

```diff
@@ -74,18 +75,28 @@ static inline void force_uaccess_end(mm_segment_t oldfs)
 static __always_inline __must_check unsigned long
 __copy_from_user_inatomic(void *to, const void __user *from, unsigned long n)
 {
+	unsigned long res;
 	instrument_copy_from_user(to, from, n);
 	check_object_size(to, n, false);
-	return raw_copy_from_user(to, from, n);
+	res = raw_copy_from_user(to, from, n);
+#ifdef CONFIG_DF_DETECTION
+	dfetch_add_address(from, n, _RET_IP_, to);
```

Comment:
For some reason I cannot answer your comment about the size being passed to dfetch_add_address().
We need to pass the actual number of bytes that were copied.
Since copy_from_user() returns the number of bytes not copied, we need to pass `n - res` to dfetch_add_address()

Reply:
ahh, okay i will fix it